src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -367,19 +196,17 @@ class InstanceSync(val project: Project, val instance: Instance) {
 
         logger.info("Installing package $remotePath on $instance")
 
-        val json = try {
-            postMultipart(url, mapOf("recursive" to config.installRecursive))
-        } catch (e: Exception) {
-            throw DeployException("Cannot install package $remotePath on $instance. Reason: request failed.", e)
-        }
-
         val response = try {
-            InstallResponse(json)
-        } catch (e: Exception) {
-            throw DeployException("Malformed install response after installing package $remotePath on $instance.", e)
-        }
+            postMultipart(url, mapOf("recursive" to config.installRecursive)) { InstallResponse.from(asStream(it), config.packageErrors) }
+        } catch (e: RequestException) {
 
-        if (!response.success) {
+            throw DeployException("Cannot install package $remotePath on $instance. Reason: request failed.", e)
+        } catch (e: ResponseException) {
+            throw DeployException("Malformed response after installing package $remotePath on $instance.")
+        }
+        if (response.hasPackageErrors) {
```

Comment:
```
val packageErrors = response.errors.filter { packageErrors.contains(it) }
if (packageErrors.isNotEmpty()) {
   throw PackageException("Cannot install package $remotePath on $install because it is malformed:\n${packageErrors.join(''\n")}")
}
```